### PR TITLE
add import restrictions for internal clients

### PR DESCRIPTION
**Pull request by deads2k**
Internal clients should not be used by any origin code other than the apiserver.  This enforces that for all groups exception builds, images, quota, routes.

@openshift/sig-security 
/assign @mfojtik

### CODE REVIEW

File: hack/import-restrictions.json [hunk 1]
```diff
@@ -70,6 +70,51 @@
...
+      "github.com/openshift/origin/pkg/network/generated",
+      "github.com/openshift/origin/pkg/oauth/generated",
+      "github.com/openshift/origin/pkg/project/generated",
+      "github.com/openshift/origin/pkg/security/generated",
+      "github.com/openshift/origin/pkg/template/generated",
+      "github.com/openshift/origin/pkg/user/generated"
+    ],
+    "allowedImportPackageRoots": [
+      "vendor",
+      ""
```
Comment (enj) [inline hack/import-restrictions.json]: Eh?

File: pkg/security/apiserver/securitycontextconstraints/matcher.go [hunk 2]
```diff
@@ -72,7 +72,6 @@ func authorizedForSCC(constraint *securityapi.SecurityContextConstraints, info u
 
 // ConstraintAppliesTo inspects the constraint's users and groups against the userInfo to determine
 // if it is usable by the userInfo.
-// TODO make this private and have the router SA check do a SAR check instead.
 // Anything we do here needs to work with a deny authorizer so the choices are limited to SAR / Authorizer
 func ConstraintAppliesTo(constraint *securityapi.SecurityContextConstraints, userInfo user.Info, namespace string, a authorizer.Authorizer) bool {
```
Comment (enj) [inline pkg/security/apiserver/securitycontextconstraints/matcher.go]: Make private?

File: pkg/oc/cli/admin/router/router.go [hunk 3]
```diff
@@ -1129,3 +1132,54 @@ func validateServiceAccount(client securityclientinternal.Interface, ns string,
 	}
 	return nil
 }
+
+// ConstraintAppliesTo inspects the constraint's users and groups against the userInfo to determine
```
Comment (enj) [inline pkg/oc/cli/admin/router/router.go]: constraintAppliesTo

File: pkg/oc/cli/admin/router/router.go [hunk 4]
```diff
@@ -1129,3 +1132,54 @@ func validateServiceAccount(client securityclientinternal.Interface, ns string,
...
+// if it is usable by the userInfo.
+// TODO remove this and have the router SA check do a SAR check instead.
```
Comment (enj) [inline pkg/oc/cli/admin/router/router.go]: Add comment to state this is copied logic

**Comment by deads2k** [comment]
comments addressed.

**Pull request merged**